dpnp/backend/kernels/dpnp_krnl_logic.cpp

```diff
@@ -29,53 +29,79 @@
 #include "dpnp_iface.hpp"
 #include "queue_sycl.hpp"
 
+
+template <typename _DataType, typename _ResultType>
+class dpnp_all_c_kernel;
+
 template <typename _DataType, typename _ResultType>
 void dpnp_all_c(void* array1_in, void* result1, const size_t size)
 {
-    _DataType* array_in = reinterpret_cast<_DataType*>(array1_in);
+    cl::sycl::event event;
+
+    const _DataType* array_in = reinterpret_cast<const _DataType*>(array1_in);
     _ResultType* result = reinterpret_cast<_ResultType*>(result1);
 
-    bool res = true;
+    result[0] = true;
 
-    for (size_t i = 0; i < size; ++i)
+    if ((array1_in == nullptr) || (result1 == nullptr))
     {
+        return;
+    }
+
+    cl::sycl::range<1> gws(size);
+    auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+        size_t i = global_id[0];
+
         if (!array_in[i])
         {
-            res = false;
-            break;
+            result[0] = false;
         }
-    }
+    };
 
-    result[0] = res;
+    auto kernel_func = [&](cl::sycl::handler& cgh) {
+        cgh.parallel_for<class dpnp_all_c_kernel<_DataType, _ResultType>>(gws, kernel_parallel_for_func);
+    };
 
-    return;
+    event = DPNP_QUEUE.submit(kernel_func);
+
+    event.wait();
 }
 
+template <typename _DataType, typename _ResultType>
+class dpnp_any_c_kernel;
+
 template <typename _DataType, typename _ResultType>
 void dpnp_any_c(const void* array1_in, void* result1, const size_t size)
 {
-    if ((array1_in == nullptr) || (result1 == nullptr))
-    {
-        return;
-    }
+    cl::sycl::event event;
 
     const _DataType* array_in = reinterpret_cast<const _DataType*>(array1_in);
     _ResultType* result = reinterpret_cast<_ResultType*>(result1);
 
-    bool res = false;
+    result[0] = false;
 
-    for (size_t i = 0; i < size; ++i)
+    if ((array1_in == nullptr) || (result1 == nullptr))
     {
+        return;
+    }
+
+    cl::sycl::range<1> gws(size);
+    auto kernel_parallel_for_func = [=](cl::sycl::id<1> global_id) {
+        size_t i = global_id[0];
+
         if (array_in[i])
         {
-            res = true;
-            break;
+            result[0] = true;
```

Comment:
I am not sure but will this work correct?
In any way bad design for parallel work.
You may need big array (size >200 for example) for the check.

@shssf why do you think it is OK? https://github.com/IntelPython/dpnp/pull/640#discussion_r588896344

Reply:
I get it too:) it will work.